nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java

```diff
@@ -184,10 +200,25 @@ protected KuduClient buildClient(final ProcessContext context) {
         final String masters = context.getProperty(KUDU_MASTERS).evaluateAttributeExpressions().getValue();
         final int operationTimeout = context.getProperty(KUDU_OPERATION_TIMEOUT_MS).evaluateAttributeExpressions().asTimePeriod(TimeUnit.MILLISECONDS).intValue();
         final int adminOperationTimeout = context.getProperty(KUDU_KEEP_ALIVE_PERIOD_TIMEOUT_MS).evaluateAttributeExpressions().asTimePeriod(TimeUnit.MILLISECONDS).intValue();
+        final int workerCount = context.getProperty(WORKER_COUNT).asInteger();
+
+        // Create Executor following approach of Executors.newCachedThreadPool() using worker count as maximum pool size
+        final int corePoolSize = 0;
+        final long threadKeepAliveTime = 60;
+        final Executor nioExecutor = new ThreadPoolExecutor(
```

Comment:
Is this required vs just setting the workerCount on the client? IIUC the client uses an unbound cached threadpool, but the NioEventLoopGroup will never user more than workerCount threads.

Reply:
Technically it isn't required as you mentioned.  However, creating the ThreadPoolExecutor instance with a custom ThreadFactory allows for created Threads to be named using the Processor Identifier.  This is useful when troubleshooting multiple instances of PutKudu in a single VM, otherwise there can be multiple threads with the same name, such as `kudu-nio-0`.